install_dependencies.sh

```diff
@@ -14,19 +14,18 @@ echo "================================================================"
 echo "Installing other ROS dependencies..."
 echo "================================================================"
 
-sudo apt-get install -y\
-    ros-kinetic-xacro \
-    ros-kinetic-controller-manager \
-    ros-kinetic-gazebo-ros-control \
-    ros-kinetic-ros-controllers \
-    ros-kinetic-ros-control \
-    ros-kinetic-effort-controllers \
-    ros-kinetic-hector-gazebo \
-    ros-kinetic-turtlebot-teleop \
-    python-rosinstall \
-    python-visual \
-    python-wxtools \
-    libserial-dev 
+# Update Rosdeps
+#rosdep update
+
+# The current directory
+CURR_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Install all required dependencies to build this repo
+rosdep install --from-paths src --ignore-src --rosdistro kinetic -y
+
+# We have this extra rosdep call because rosdep can't seem to find nested
+# `package.xml` files, like those in src/gps_umd/gps_common
+rosdep install --from-paths src/gps_umd --rosdistro kinetic -y
```

Comment:
This switches us over from using a list of packages to install to using rosdep's, which basically check the dependencies listed in each `package.xml` file and installs them. It has some issues if they're deeply nested (as is the case with those in `gps_umd`) but otherwise this is a much better system that should make sure we can keep dependencies updated without having to worry about maintaining a list. This should also resolve #94!